python/cudf/cudf/core/indexing.py

```diff
@@ -164,10 +165,19 @@ def __setitem__(self, key, value):
         self._sr.iloc[key] = value
 
     def _loc_to_iloc(self, arg):
-        from cudf.core.column import column
-        from cudf.core.series import Series
-
         if is_scalar(arg):
+            if not is_numerical_dtype(self._sr.index.dtype):
+                # TODO: switch to cudf.utils.dtypes.is_integer(arg)
+                if pd.api.types.is_integer(arg):  #
+                    found_index = arg
+                    return found_index
+
+                elif isinstance(
+                    arg, cudf.Scalar
+                ) and pd.api.types.is_integer_dtype(arg.dtype):
+                    found_index = arg.value
+                    return found_index
```

Comment:
I think we should swap the order of these. I'm not sure if `pd.api.types.is_integer(...)` can handle a `cudf.Scalar` object, but generally we want the `cudf.Scalar` route to be the fastest if possible.